smart-lock

```diff
@@ -540,16 +555,17 @@ function android-compile {
     #    sed -e "s/#SPDP_RTPS_RELAY_ADDRESS_LINE/SpdpRtpsRelayAddress=${RTPS_RELAY_IP}:4444/g" -e "s/#SEDP_RTPS_RELAY_ADDRESS_LINE/SedpRtpsRelayAddress=${RTPS_RELAY_IP}:4445/g" -e "s/#DATA_RTPS_RELAY_ADDRESS_LINE/DataRtpsRelayAddress=${RTPS_RELAY_IP}:4446/g" -i'' "${_SMART_LOCK_DIR}/android/SmartLock/app/src/main/assets/opendds_config.ini"
     #fi
 
-    docker run --rm -ti -v "${_SMART_LOCK_DIR}:/home/droid/smartlock" -w '/home/droid' android-opendds-cross bash -c '
+    ${DOCKER_COMMAND} run --rm -ti -v "${_SMART_LOCK_DIR}:/home/droid/smartlock" -w '/home/droid' android-opendds-cross bash -c '
 APP_MPC_NAME=SmartLock_Idl_Java;
 APP_NAME=SmartLock;
 DDS_ROOT=/home/droid/droid-opendds;
 source $DDS_ROOT/build/target/setenv.sh;
-echo $HOME
 cd $HOME/smartlock/android/Idl;
+sudo chmod -R o+w ..;
+mwc.pl -type gnuace;
+make -f GNUmakefile.${APP_MPC_NAME} ${APP_NAME}TypeSupportJC.h;
```

Comment:
Are these still needed?

Reply:
Yes.  We need to ensure that the TypeSupport.idl gets generated.  And, then we have to re-generate (the step after that make call) to get the makefiles to contain all of the rules for all of the .class files determined by parsing the TypeSupport.idl file.

Reply:
When we document how users should set up Java-enabled OpenDDS projects we don't have anything indicating that this would be expected (same for the in-repo tests).  So I was thinking that the "helper" Perl modules would be sufficient to generate a makefile (the first time we generate) that has all the rules to generate needed classes.

Reply:
I'm not sure what's different.  But, without `SmartLockTypeSupport.idl` present, the generated makefile is missing 60 rules for building .class files.

Reply:
If you look near the end of this log, you will see the error that we get when the makefile is generated without `SmartLockTypeSupport.idl` present.

https://pipelines.actions.githubusercontent.com/serviceHosts/0369b20c-e68f-4ff7-b7c1-ff38dd11af7c/_apis/pipelines/1/runs/14/signedlogcontent/6?urlExpires=2023-03-09T15%3A32%3A27.2679016Z&urlSigningMethod=HMACV1&urlSignature=xNpWJ%2FnkUivklEowkWmYTT%2BBn0RpX653GVKYBBhYA%2FE%3D